typing/typecore.ml

```diff
@@ -908,7 +908,28 @@ let type_continuation_pat env expected_ty sp =
           Types.val_loc = loc; val_attributes = []; }
       in
         Some (id, desc)
-  | Ppat_extension ext ->
+  | Ppat_constraint ({ppat_desc=Ppat_var name; ppat_loc=lloc},
```

Comment:
@lpw25 Here, I have restricted `ppat_desc` to just `Ppat_var`. Am I missing any other valid cases?